packages/cli/src/tokens/build.ts

```diff
@@ -6,125 +6,99 @@ import chalk from 'chalk';
 import * as R from 'ramda';
 import StyleDictionary from 'style-dictionary';
 
-import * as configs from './build/configs.js';
+import { getConfigsForThemeDimensions } from './build/configs.js';
+import type { BuildConfig, ThemePermutation } from './build/types.js';
 import { makeEntryFile } from './build/utils/entryfile.js';
 
-const { permutateThemes, getConfigs } = configs;
-
 type Options = {
-  /** Design tokens path  */
+  /** Design tokens path */
   tokens: string;
-  /** Output directoru for built tokens */
+  /** Output directory for built tokens */
   out: string;
   /** Generate preview tokens */
   preview: boolean;
   /** Enable verbose output */
   verbose: boolean;
 };
 
-// type FormattedCSSPlatform = { css: { output: string; destination: string }[] };
+export let buildOptions: Options | undefined;
 
 const sd = new StyleDictionary();
 
+/*
+ * Declarative configuration of the build output
+ */
+const buildConfigs = {
+  typography: { sdConfig: 'typographyVariables', dimensions: ['typography'] },
+  'color-mode': { sdConfig: 'colorModeVariables', dimensions: ['mode'] },
+  semantic: { sdConfig: 'semanticVariables', dimensions: ['semantic'] },
+  storefront: {
+    name: 'Storefront preview tokens',
+    sdConfig: 'typescriptTokens',
+    dimensions: ['mode'],
+    options: { outPath: path.resolve('../../apps/storefront/tokens') },
+  },
+  entryFiles: {
```

Comment:
I would suggest separating this from the build config to avoid another `buildAllPlatforms` call as all it needs is just a list of theme names, which can be extracted elsewhere and iterated over?

Reply:
There is no (implicit) buildAllPlatforms when this config is run, as the custom `build` function is run instead. See early return around line 90:
```ts
if (buildConfig.build) {
    return await buildConfig.build(sdConfigs, { outPath, tokensDir, ...buildConfig.options });
}
```

Reply:
Ah, your right. I missed that return there :)